### PR TITLE
Initial version of Docker cleanup image

### PR DESCRIPTION
**Pull request by danielgrant**
Consists of:

- A bash script which executes, inside a loop, `docker` commands to clean up stopped containers, images with no associated container, and unused volumes.
- A Dockerfile that bundles the bash script.
- Linting, building and pushing using pre-commit and TravisCI

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+install:
+  - pip install pre-commit
+
+script:
+  - 'make lint'
+  - 'make image'
+
+after_success:
+  - if [[ "$TRAVIS_BRANCH" == "master" ]]; then
```
Comment (stephenwf) [inline .travis.yml]: I think we need to check that we're not also on a pull request as the branch will be master on the "PR" build in travis.
```
if [[ "$TRAVIS_BRANCH" = "master" ]] && [[ "$TRAVIS_PULL_REQUEST" = "false" ]]; then
```

Also noticed:
```
/home/travis/.travis/functions: eval: line 110: syntax error: unexpected end of file
```
[here](https://travis-ci.com/github/digirati-co-uk/docker-cleanup/builds/198472082#L350) as it tries to run, not sure what's caused that though

Comment (danielgrant) [reply .travis.yml]: Good spot, fixed that now so it won't push on PR's.

I believe the syntax error was due to the absence of semicolons at the end of the statements inside the if-statement. Fixed now.

**Review by stephenwf — APPROVED** [review]
Neat tool!

**Comment by stephenwf** [comment]
Tried it out, following the docs and worked like a charm.

**Comment by danielgrant** [comment]
> Tried it out, following the docs and worked like a charm.

Thanks for testing it out @stephenwf - I'm going to drop it into a Kubernetes cluster soon and then share the manifest as part of the repo.

**Pull request merged**